### PR TITLE
Check if existing sort order equals new sort order.

### PR DESCRIPTION
**Pull request by caalador**
If they equal do not do anything for sort request.

---
This change is [Reviewable](https://reviewable.io/reviews/vaadin/vaadin-grid-flow/80)

### CODE REVIEW

**Comment by denis-anisimov** [comment]
Reviewed 1 of 1 files at r1.
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---

*[a discussion](https://reviewable.io:443/reviews/vaadin/vaadin-grid-flow/80#-L4LFl00r5uf4ZMjfIlR:-L4LFl00r5uf4ZMjfIlS:btq11jc) (no related file):*
Unit test ?

---

*[src/main/java/com/vaadin/flow/component/grid/Grid.java, line 1726 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-grid-flow/80#-L4LFVpjjSIB5f3KA_jN:-L4LFVpjjSIB5f3KA_jO:buin58t) ([raw file](https://github.com/vaadin/vaadin-grid-flow/blob/e25e91eea2d16cbde6aa026d33672f6ac9132e3c/src/main/java/com/vaadin/flow/component/grid/Grid.java#L1726)):*
>  private static boolean listsAreEqual(final List<?> list1,
>             final List<?> list2) {

Do you really need this method ?

`List:equals` defines the contract that you need I think:
```
In other words, two lists are defined to be
 equal if they contain the same elements in the same order. 
```

---

*[src/main/java/com/vaadin/flow/component/grid/Grid.java, line 1742 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-grid-flow/80#-L4LF6DjSjIeI_Yjg_OE:-L4LF6DjSjIeI_Yjg_OF:b-snlru8) ([raw file](https://github.com/vaadin/vaadin-grid-flow/blob/e25e91eea2d16cbde6aa026d33672f6ac9132e3c/src/main/java/com/vaadin/flow/component/grid/Grid.java#L1742)):*
>             if (!(obj1 == null ? obj2 == null : obj1.equals(obj2))) {
>                 return false;
>             }

`Objects.equals` ?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-grid-flow/80)*

**Comment by caalador** [comment]
Review status: 0 of 3 files reviewed at latest revision, 3 unresolved discussions.

---

*[a discussion](https://reviewable.io:443/reviews/vaadin/vaadin-grid-flow/80#-L4LFl00r5uf4ZMjfIlR:-L4LK5uprYm9Ce2ctbHa:b-7tfmjp) (no related file):*
<details><summary><i>Previously, denis-anisimov (Denis) wrote…</i></summary><blockquote>

Unit test ?
</blockquote></details>

added

---

*[src/main/java/com/vaadin/flow/component/grid/Grid.java, line 1726 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-grid-flow/80#-L4LFVpjjSIB5f3KA_jN:-L4LK6vVki1DIj-vW9cR:b-896fix) ([raw file](https://github.com/vaadin/vaadin-grid-flow/blob/e25e91eea2d16cbde6aa026d33672f6ac9132e3c/src/main/java/com/vaadin/flow/component/grid/Grid.java#L1726)):*
<details><summary><i>Previously, denis-anisimov (Denis) wrote…</i></summary><blockquote>

>  private static boolean listsAreEqual(final List<?> list1,
>             final List<?> list2) {

Do you really need this method ?

`List:equals` defines the contract that you need I think:
```
In other words, two lists are defined to be
 equal if they contain the same elements in the same order. 
```
</blockquote></details>

Done.

---

*[src/main/java/com/vaadin/flow/component/grid/Grid.java, line 1742 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-grid-flow/80#-L4LF6DjSjIeI_Yjg_OE:-L4LK7MTVynpT4TMpAUt:b-896fix) ([raw file](https://github.com/vaadin/vaadin-grid-flow/blob/e25e91eea2d16cbde6aa026d33672f6ac9132e3c/src/main/java/com/vaadin/flow/component/grid/Grid.java#L1742)):*
<details><summary><i>Previously, denis-anisimov (Denis) wrote…</i></summary><blockquote>

>             if (!(obj1 == null ? obj2 == null : obj1.equals(obj2))) {
>                 return false;
>             }

`Objects.equals` ?
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-grid-flow/80)*
